pandas/core/series.py

```diff
@@ -2292,10 +2303,22 @@ def combine(self, other, func, fill_value=None):
         1    2
         dtype: int64
```

Comment:
Can you replace this example by one with the number of legs and arms of couple of animals, and combining them using the `sum` function instead. I think somehow real examples are much easier to understand. One Series can have 3 values and the other two, so we just need the basic example, and then one with `fill_value=0`.

Reply:
> Can you replace this example by one with the number of legs and arms of couple of animals, and combining them using the sum function instead

This returns error, @datapythonista 
```
>>> s1 = pd.Series([2,0],[1,2])
>>> s2 = pd.Series([0,2,2])
>>> s1
1    2
2    0
dtype: int64
>>> s2
0    0
1    2
2    2
dtype: int64
>>> s1.combine(s2,sum)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/series.py", line 2240, in combine
    new_values[i] = func(lv, rv)
TypeError: 'float' object is not iterable
>>> s1.combine(s2,lambda x,y:x+y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/series.py", line 2240, in combine
    new_values[i] = func(lv, rv)
ValueError: cannot convert float NaN to integer
```

Reply:
Please see this, @datapythonista 
```
>>> s1 = pd.Series([2,0],[1,2])
>>> s2 = pd.Series([0,2,2])
>>> s1
1    2
2    0
dtype: int64
>>> s2
0    0
1    2
2    2
dtype: int64
>>> s1.combine(s2,sum)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/series.py", line 2240, in combine
    new_values[i] = func(lv, rv)
TypeError: 'float' object is not iterable
>>> s1.combine(s2,lambda x,y:x+y)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/series.py", line 2240, in combine
    new_values[i] = func(lv, rv)
ValueError: cannot convert float NaN to integer
>>> s1.combine(s2,sum,fill_value=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/dist-packages/pandas/core/series.py", line 2240, in combine
    new_values[i] = func(lv, rv)
TypeError: 'int' object is not iterable
>>> s1.combine(s2,lambda x,y:x+y,fill_value=0)
0    0
1    4
2    2
dtype: int64
>>>
```

Reply:
I've pushed the other changes as requested, @datapythonista 
Please help me with the example, I'm finding it difficult to understand what you meant,

Reply:
if you do a grep in the pandas repo looking for falcon, bear or penguin, you'll find some examples where instead of using foo, bar, 1, 2, 3... we use something more meaningful, that in my opinion help users understand better what the operations are doing. My comment was that we should do the same here.

Reply:
can you update this?

Reply:
working on it atm
is this okay, @datapythonista ?
```
>>> arms = pd.Series([2, 2, 2])
>>> legs = pd.Series([2, 2])
>>> limbs = arms.combine(legs, lambda x1,x2: x1+x2, fill_value=0)
>>> limbs
0    4
1    4
2    2
dtype: int64
>>>
```